assets/model_monitoring/components/src/shared_utilities/store_url.py

```diff
@@ -96,20 +97,31 @@ def get_credential(self) -> Union[
                 return ClientSecretCredential(tenant_id=self._datastore.tenant_id, client_id=self._datastore.client_id,
                                               client_secret=self._datastore.client_secret)
             else:
+                from azure.ai.ml.identity import AzureMLOnBehalfOfCredential
                 return AzureMLOnBehalfOfCredential()
         else:
             raise InvalidInputError(f"Unsupported datastore type: {self._datastore.datastore_type}, "
                                     "only Azure Blob and Azure Data Lake Gen2 are supported.")
 
     def is_credentials_less(self) -> bool:
         """Check if the store url is credential less."""
-        credential = self.get_credential()
+        # TODO: remove after we figure out cache failure issues.
+        # Should be able to import AzureMLOnBehalfOfCredential to check the class w/o issues.
+        credential = None
+        try:
+            credential = self.get_credential()
```

Comment:
can you print some log for enabling credentialess path and use the managed identity successfully (or maybe we already have that in your first PR)?  Because it can help us debug our managed identity works as expected.

Reply:
I added some log statements when we return the credential to know what type it is, this should let us know if the StoreUrl is using credential-less path.

Also, from the first PR I added a helpful error message that is printed when the read/write fails and it is due to CredentialUnavailableError which afaik means the managed identity is not attached or the job was not submitted with "managed".

Otherwise, if the credential is OBO token and there are no errors saving/writing data then the momo job is using managed identity and is working as expected I think.